internetarchive/openlibrary
# `i18n` syntax additions with variables

Subtask of #9486.


Fix. This PR adds `i18n` syntax in cases where the text to be translated includes a variable or variables. See the i18n [guide](https://github.com/internetarchive/openlibrary/wiki/Internationalization#html-files) for full syntax info and instructions.

### Technical

Separated the PR's by `i18n` syntax type for ease of review, this one is scoped to cover all the `i18n` variable syntax.

Also includes removing relevant files from `i18n` exclude list so as to confirm they now pass the `detect-missing-i18n` check.

### Testing

1. Find a newly `i18n`-ed phrase in one of the files that includes a variable
2. Confirm that the new variable is assigned to the correct variable name, i.e. `View $num_works works` should become `$_('View %(count)d works', count=num_works)`
3. Confirm that the new variable uses the correct format, i.e. `%( )s` if it will be assigned to a string, vs. `%( )d` if it will be assigned to a number
4. Check to see that the phrase appears correctly in the `messages.pot` file
5. Run the `i18n` detection script with `pre-commit run detect-missing-i18n  --all-files`
6. No errors or warnings should appear, and the check should pass

### Screenshot


### Stakeholders

@mekarpeles @cdrini @pidgezero-one
